homeassistant/components/ipma/weather.py

```diff
@@ -103,6 +122,23 @@ def __init__(self, station, config):
             self._forecast = await self._station.forecast()
             self._description = self._forecast[0].description
 
+    @property
+    def unique_id(self) -> str:
+        """Return a unique id."""
+        return '{}, {}'.format(self._station.latitude, self._station.longitude)
+
+    @property
+    def device_info(self):
+        """Return device information."""
+        return {
+            'identifiers': {
+                (DOMAIN, self.unique_id)
+            },
+            'name': self.name,
+            'manufacturer': "Instituto Português do Mar e Atmosfera",
```

Comment:
This isn't really a manufacturer. I would call it a service provider. We don't have a key for that yet.

Reply:
I actually asked Paulus on discord about this... 

I asked because I had myself doubts about if it would make sense - it's a service not a device.

I nonetheless see usefulness to show the user this information.

Reply:
Was this the outcome of the discussion?

I don't think we can add arbitrary keys unfortunately, although I think that would be good. I think it's weird if we add values that don't match the key names, but it's not the end of the world for me, if that's what we should do now.

Reply:
The outcome was that it was OK for a weather platform to use the device registry.

I personally interpreted manufacturer -> provider and device -> service.

Reply:
Maybe we need to have clarification what is supported technically. As far as I have read the code, there's a set of keys that used. So using different keys without adding support for that is a no show. But I'm not 💯.

Reply:
I don't want to make an issue out of this, so I'm dropping the support.